enum.go

```diff
@@ -0,0 +1,85 @@
+package wf
+
+import (
+	"golang.org/x/sys/windows"
+)
+
+type RuleEnumerator struct {
+	session      *Session
+	enumTemplate fwpmFilterEnumTemplate0
+}
+
+func (e RuleEnumerator) WithProvider(provider ProviderID) RuleEnumerator {
+	e.enumTemplate.ProviderKey = new(ProviderID)
+	*e.enumTemplate.ProviderKey = provider
```

Comment:
out of curiosity, in this context, is the intention here to get a copy of the ProviderID vs. just storing a reference to the provider passed in, in case the consumer changes it?

Reply:
Sadly, this is just the interface provided by Microsoft. It needs to be a GUID pointer.

Reply:
Similarly, changing `CalloutKey` to the `CalloutID` type should allow the same change in `WithCalloutKey` as well.